### PR TITLE
Chat: Make HTML pages display an error on crash

### PR DESCRIPTION
**Pull request by mia-pi-git**
Currently, they just don't do anything, and it's difficult to tell if it's crashed for some.

### CODE REVIEW

**Comment by Zarel** [comment]
I think the real question is why they normally don't. Can you show me an example crash message this generates? That would probably help tell us what's wrong.

**Comment by mia-pi-git** [comment]
> I think the real question is why they normally don't. Can you show me an example crash message this generates? That would probably help tell us what's wrong.

```CRASH: Error: test
    at PageContext.test (/Users/mia/Documents/GitHub/pokemon-showdown-dev/.server-dist/chat-plugins/private/test.js:3:9)
    at PageContext.resolve (/Users/mia/Documents/GitHub/pokemon-showdown-dev/.server-dist/chat.js:268:36)
    at _class.resolvePage (/Users/mia/Documents/GitHub/pokemon-showdown-dev/.server-dist/chat.js:1998:56)
    at User.tryJoinRoom (/Users/mia/Documents/GitHub/pokemon-showdown-dev/.server-dist/users.js:1272:16)
    at CommandContext.join (/Users/mia/Documents/GitHub/pokemon-showdown-dev/.server-dist/chat-commands/moderation.js:320:15)
    at CommandContext.run (/Users/mia/Documents/GitHub/pokemon-showdown-dev/.server-dist/chat.js:525:28)
    at CommandContext.parse (/Users/mia/Documents/GitHub/pokemon-showdown-dev/.server-dist/chat.js:357:19)
    at _class.parse (/Users/mia/Documents/GitHub/pokemon-showdown-dev/.server-dist/chat.js:1450:18)
    at User.chat (/Users/mia/Documents/GitHub/pokemon-showdown-dev/.server-dist/users.js:1425:9)
    at Object.socketReceive (/Users/mia/Documents/GitHub/pokemon-showdown-dev/.server-dist/users.js:1680:12)

Additional information:
  user = mia ⚢ pi
  room = global
  pageid = view-test```

File: server/chat.ts [hunk 1]
```diff
@@ -263,7 +263,21 @@ export class PageContext extends MessageContext {
 		parts.shift();
 		while (handler) {
 			if (typeof handler === 'function') {
-				let res = await handler.bind(this)(parts, this.user, this.connection);
+				let res;
+				try {
+					res = await handler.bind(this)(parts, this.user, this.connection);
```
Comment (Zarel) [inline server/chat.ts:268]: Looks like Ashe forgot about `call` existing.

```suggestion
					res = await handler.call(this, parts, this.user, this.connection);
```

**Comment by Zarel** [comment]
Oh! I understand the problem now. Yes, this is a great idea.

**Pull request merged**